test/test-exception.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+// Testcase to produce NodeReport on uncaught exception
+if (process.argv[2] === 'child') {
+  require('../').setEvents("exception");
+
+  function myException(request, response) {
+    throw new UserException('*** exception.js: testcase exception thrown from myException()');
+  }
+
+  function UserException(message) {
+    this.message = message;
+    this.name = "UserException";
+  }
+
+  myException(); 
+} else {
+  const tap = require('tap');
+  const common = require('./common.js');
+  const fs = require('fs');
+  const spawn = require('child_process').spawn;
+
+  var child = spawn(process.execPath, [__filename, 'child']);
+  child.on('exit', function(code, signal) {
+    tap.plan(3);
+    tap.notEqual(code, 0, 'Process should not exit cleanly');
```

Comment:
Re "Guess this is a bug" - assuming this is a reference to` Illegal instruction (core dumped)`

That's the v8 behaviour when --abort-on-uncaught-exception is set, and an exception is thrown and not caught. You can see it without nodereport interference by running:
    `node --abort-on-uncaught-exception unknown.js`  
v8::base::OS::Abort() intentionally aborts and triggers a core dump using an illegal instruction. It's what they call a 'hard' abort. Not sure why, in the JVM we use `raise(SIGABRT)`, but maybe there are some situations where that gets intercepted. 

However, see also discussion in https://github.com/nodejs/nodereport/issues/6 We need to change the default nodereport behaviour, currently it adds the --abort-on-uncaught-exception flag under the covers, and the user has to switch that off via the nodereport.setCoreDump() APi if they want normal Node.js behaviour.